README.md

```diff
@@ -11,80 +10,54 @@ package that ships with GHC.
 
 The purpose of `linear-base` is to provide the minimal facilities you need to
 write _practical_ Linear Haskell code, i.e., Haskell code that uses the
-`-XLinearHaskell` language extension.
+`-XLinearTypes` language extension.
 
 ## Motivation
 
 _Why do you need `linear-base` to write linear projects?_
 
-1. If you are writing a linear function, you cannot use the standard
-  non-linear data types, functions and classes in `base`. Even simple uses of
-  `base` facilities break down. For instance, if `n` is a linearly bound `Int`,
-  the RHS of an definition cannot write `n + 1` --- this will not type check. We need
-  linear variants of `Num`, `Functor`s, `Monad`s, `($)`, etc.
-
-2. There are several primitive data structures, functions and classes
-  that can only be safely released with `-XLinearHaskell`.  For example, the
-  API for file IO released by `linear-base` is safe unlike the current API
-  which allows, say, reading from a closed file handle.  _Some of these are new
-  and exciting because they are not in `base`, e.g., `linear-base` releases a
-  safe API for mutable arrays allowing values to be updated in place!_
-
-## Getting Started
-
-Since -XLinearTypes are not part of a GHC release, we need to use nix to get a
-version of GHC with support for linear types.
-
-1. In your stack project, add our [`shell-stack.nix`] and [`nixpkgs.nix`] files.
-2. Edit your `stack.yaml` to have the following fields.  _Use an appropriate
-   commit in the `extra-deps`._
-
-```yaml
-resolver: lts-16.2
-compiler: ghc-8.11
-allow-newer: true
-system-ghc: true
-
-nix:
-  enable: true
-  shell-file: shell-stack.nix
-  path: ["nixpkgs=./nixpkgs.nix"]
-
-extra-deps:
-  - git: https://github.com/facundominguez/quickcheck.git
-    commit: a498e7b41131cf7955b9e154ab26d37d1be10304
-  - git: https://github.com/facundominguez/lifted-async.git
-    commit: 898eb485b21cc321058345998eedd8c409c326fd
-  - git: https://github.com/tweag/linear-base.git
-    commit: 705522f0bad3f1083bb3447b9f476f6f84d21410
-```
+1. Data types, functions and classes in `base` are not linear types
+  aware. For instance, if `n` is a linearly-bound `Int`, the RHS of
+  a definition cannot write `n + 1` — this will not type check. We
+  need linear variants of `Num`, `Functor`s, `Monad`s, `($)`, etc.
+
+2. This library exports new abstractions that leverage linear types
+  for resource safety or performance. For example, there are new APIs
+  for file and socket I/O as well as for safe in-place mutation of
+  arrays.
+
+## Getting started
 
-4. Add `linear-base` to your cabal file's `build-depends:` for the appropriate
-   library or executable.
+`-XLinearTypes` is not yet part of any GHC release. We recommend using
+Stack and Nix together to pull in an experimental version of GHC into
+your project. Use [this `stack.yaml`](./stack.yaml) as a starting
+point for your project. To learn more about Stack+Nix integration, see
+[here](https://docs.haskellstack.org/en/stable/nix_integration/).
```

Comment:
@mboes It seems to me this is not friendly to new users. If I want to try playing with linear types and linear base, I shouldn't have to learn about using Stack and Nix together. I should be able to follow some simple instructions, install nix and stack if I have to and then just start coding. If putting a lengthy getting started section here isn't a good idea, I could make a template repository with minimal instructions there and link to it from here.

Reply:
Somebody recently reached out to me about how to use `LinearTypes` without using Nix. Other than compiling GHC manually, it seems like using `ghcup` with "custom bindists" (https://gitlab.haskell.org/haskell/ghcup-hs#installing-custom-bindists) is another option. So I was wondering if we should mention other methods here as well.

However, we are supposed to get a proper GHC release within a month or so (https://gitlab.haskell.org/ghc/ghc/-/wikis/status/ghc-9.0.1); so I personally don't worry much about making this section as beginner friendly as possible.

Reply:
It's really not worth pursuing: we provide one preferred avenue that we know works (because we are testing it). Using an unreleased version of GHC is a power user thing, anyway, so we can assume that users will figure it out.

Reply:
> I shouldn't have to learn about using Stack and Nix together

As @aspiwack says above, yes you should, if to start with you're willing to use an unreleased library that depends on an unreleased compiler featuring an experimental extension. As maintainers, we're not on the hook for providing multiple ways to build the library, with the risk of each system going out of sync with other because we're not testing or we are testing but the CI state space explodes. We're only on the hook for making sure *some* simple command exists that will reliably build the whole thing and execute the tests, and for making really sure that command does work.

Once GHC 9.0 ships, we should strive to have the library included in Stackage. From that point onwards, releases can be built like any other Cabal package. Releases should be frequent. And it will still be fair to support only one command for building and testing the `master` branch, for power users.

Reply:
Hmm, I see why that puts too much on maintainers. In that case, I'll agree that it's important to put it on Stackage. It's much easier to make linear types mainstream if there's a low/non-existant barrier to entry.